src/server/cluster/cluster_family.cc

```diff
@@ -718,12 +722,71 @@ void ClusterFamily::MigrationConf(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  cntx->conn()->SetName("slot_migration_ctrl");
+  auto sync_id = CreateMigrationSession(cntx, port);
 
+  cntx->conn()->SetName("slot_migration_ctrl");
+  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  rb->StartArray(2);
   cntx->SendLong(shard_set->size());
+  cntx->SendLong(sync_id);
   return;
 }
 
+uint32_t ClusterFamily::CreateMigrationSession(ConnectionContext* cntx, uint16_t port) {
+  std::lock_guard lk(migration_mu_);
+  auto sync_id = next_sync_id_++;
+  auto info = make_shared<MigrationInfo>(cntx->conn()->RemoteEndpointAddress(), sync_id, port);
+  auto [it, inserted] = migration_infos_.emplace(sync_id, info);
+  CHECK(inserted);
+  return sync_id;
+}
+
+void ClusterFamily::Flow(CmdArgList args, ConnectionContext* cntx) {
+  CmdArgParser parser{args};
+  auto [sync_id, shard_id] = parser.Next<uint32_t, uint32_t>();
+  DCHECK(!parser.Error());
+
+  VLOG(1) << "Create flow "
+          << " sync_id: " << sync_id << " shard_id: " << shard_id << " shard";
+
+  cntx->conn()->SetName(absl::StrCat("migration_flow_", sync_id));
+
+  auto info = GetMigrationInfo(sync_id);
+  if (!info)
+    cntx->SendError(kIdNotFound);
+
+  info->conn = cntx->conn();
```

Comment:
I dont follow how this is working.
We have sync id per migration which is created on the  DFLYMIGRATE conf which is running from the main migration connection
Then you call the DFLYMIGRATE flow command which runs on all fibers one for each shard.
How come you call the GetMigrationInfo for the sync_id and every time you update the conn var with the cntx->conn() which from what I understand just overides the value every time you call this. What am I missing?

Reply:
good catch